### PR TITLE
begin implementing the kybra extension

### PR DESCRIPTION
**Pull request by lastmjs**
Closes https://github.com/demergent-labs/kybra/issues/486

Adding the Kybra extension just like in Azle. This greatly simplifies the dfx.json file.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -210,6 +207,12 @@ jobs:
               shell: bash -l {0}
               working-directory: ${{ matrix.example_directories }}
               run: source venv/bin/activate && pip install -r requirements.txt
+            - if: ${{ needs.release-candidate-deploy.outputs.should_run_tests }}
+              working-directory: ${{ matrix.example_directories }}
+              run: source venv/bin/activate && python -m kybra install-dfx-extension
```
Comment (lastmjs) [inline .github/workflows/test.yml:212]: Same question for me as in Azle, should we automatically install this? I'm waiting on some guidance for this, we might be able to do this automatically soon...we just need to be able to run dfx start in the same directory as a dfx.json file with an unsupported extension

File: kybra/__main__.py [hunk 2]
```diff
@@ -118,7 +126,7 @@ def create_paths(args: Args) -> Paths:
     py_file_names_file_path = f"{canister_path}/py_file_names.csv"
 
     # This is the path to the developer's Candid file passed into python -m kybra from the dfx.json build command
-    did_path = args["did_path"]
+    did_path = os.environ.get("CANISTER_CANDID_PATH", "CANDID_PATH_NOT_FOUND")
```
Comment (lastmjs) [inline kybra/__main__.py:129]: Should we throw if this is not found?

File: kybra/__main__.py [hunk 3]
```diff
@@ -20,6 +20,15 @@
 
 @timed
 def main():
+    if sys.argv[1] == "install-dfx-extension":
```
Comment (lastmjs) [inline kybra/__main__.py:23]: Should I comment that this is temporary? Or explain that a more official way to install is coming?

File: kybra/compiler/dfx_extension/extension.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{
+    "name": "kybra",
+    "version": "0.0.0",
```
Comment (lastmjs) [inline kybra/compiler/dfx_extension/extension.json:3]: How should versioning work? If this matches the version of kybra then we'll never have confusion about which version is which...we should get this to update automatically then in our publish script or something

File: kybra/compiler/dfx_extension/extension.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    "homepage": "https://github.com/dfinity/dfx-extensions",
+    "authors": "",
+    "summary": "",
+    "categories": [],
+    "keywords": [],
```
Comment (lastmjs) [inline kybra/compiler/dfx_extension/extension.json:8]: update this metadata later, make an issue for it here and in azle

File: kybra/compiler/dfx_extension/install.sh [hunk 6]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+DFX_CACHE_DIR="$(dfx cache show)"
+dfx cache install
```
Comment (lastmjs) [inline kybra/compiler/dfx_extension/install.sh:4]: Do we need to run dfx cache install? It erases any other unofficial extensions

File: kybra/__main__.py (hunk 3 above)
Comment (lastmjs) [reply kybra/__main__.py:23]: Comment added

File: kybra/__main__.py (hunk 2 above)
Comment (bdemann) [reply kybra/__main__.py:132]: Maybe, lets talk about it

**Review by bdemann — CHANGES_REQUESTED** [review]

File: kybra/__main__.py (hunk 2 above)
Comment (lastmjs) [reply kybra/__main__.py]: Done

**Pull request merged**